Doc/library/dataclasses.rst

```diff
@@ -406,6 +406,11 @@ Module contents
    of the dataclass is set to that value.
    By default, it is set to the module name of the caller.
 
+   The *dataclass_factory* is a callable that will be used to create the dataclass.
+   It should take the class object as a first argument and the same keyword arguments
```

Comment:
```suggestion
   It should take the class object as a first parameter and the same keyword parameters
```

Let me know if I should make the change.

Reply:
Here, you don't need to. In `def add(x, y)`, the names "x" and "y" are the parameters of the function "add", whereas `add(3, 5)` illustrates a call to the function `add` with arguments 3 and 5. In your case, you could say that the first parameter is expected to be a class object but you could also say that the first argument of the function call would be the class object. In the docs, we also are not really consistent in the distinction between parameters and arguments (we sometimes say "The *x* argument [...]" or "The *x* parameter is [...]" or "The parameter must be not [...]" or "The argument must not [...]".

The reason why I changed your NEWS entry is because you added a new parameter, not an argument (well, some people refer to parameters as formal arguments and arguments as actual arguments, but I prefer distinguishing them two by reserving the word argument for actual arguments and parameters as parameters or formal parameters to stress that it's a formal one).

Reply:
Thanks for the detailed explanation, I appreciate the effort put in these kind of details in docs!